File: ojscala/src/main/scala/com/zhourui/leetcode/lc0039_combinationsum.scala
package com.zhourui.leetcode
import scala.util.control.Breaks._
import scala.collection.mutable.Stack

package lc0039_combinationsum {
  object Solution {
    def combinationSum(candidates: Array[Int], target: Int): List[List[Int]] = {
      var arr = candidates
      scala.util.Sorting.quickSort(arr)
      var ans = Vector[List[Int]]()
      var subset  = Stack[Int]()

      def helper(nums:Array[Int], start:Int, rest:Int): Unit = {
        if (rest == 0) {
          ans = ans :+ subset.toList
          return
        }
        breakable {
          for (i<- start until nums.length) {
            if (rest>=nums(i)) {
              subset.push(nums(i))
              helper(nums,i,rest-nums(i))
              subset.pop
            } else {
              break()
            }
          }
        }
      }
      helper(arr,0,target)
      ans.toList
    }
  }
}

File: ojjava/src/main/java/com/zhourui/leetcode/Lc0039_combinationsum.java
package com.zhourui.leetcode;

import com.zhourui.codech.BaseSolution;

import java.util.List;

public class Lc0039_combinationsum extends BaseSolution {
    class Solution {
        public List<List<Integer>> combinationSum(int[] candidates, int target) {

        }

        private void dfs(List<List<Integer>> ret,List<Integer> comb,int start,int[] cand, int target) {
            if (target==0) {
                ret.add(comb);
                return;
            }
            for (int i=start; i< cand.length;i++) {
                
            }
        }
    }
}
